livre/lib/cache/TextureDataObject.cpp

```diff
@@ -29,213 +30,222 @@
+           case DT_INT8:
+           {
+                const int8_t* data = static_cast< const int8_t* >( rawData );
+                const Vector3f min( std::numeric_limits< int8_t >::min( ));
+                const Vector3f max( std::numeric_limits< int8_t >::max( ));
+                signedQuantize( data, &formattedData[ 0 ], dataSize,
+                                compCount, min, max );
+                break;
+           }
+           case DT_INT16:
+           {
+                const int16_t* data = static_cast< const int16_t* >( rawData );
+                const Vector3f min( std::numeric_limits< int16_t >::min( ));
+                const Vector3f max( std::numeric_limits< int16_t >::max( ));
+                signedQuantize( data, &formattedData[ 0 ], dataSize,
+                                compCount, min, max);
+                break;
+           }
+           case DT_INT32:
+           {
+                const int32_t* data = static_cast< const int32_t* >( rawData );
+                const Vector3f min( std::numeric_limits< int32_t >::min( ));
+                const Vector3f max( std::numeric_limits< int32_t >::max( ));
+                signedQuantize( data, &formattedData[ 0 ], dataSize,
+                                compCount, min, max );
+                break;
+           }
+           case DT_UNDEFINED:
+           case DT_FLOAT32:
+           case DT_FLOAT64:
+           {
+                LBTHROW( std::runtime_error( "Unimplemented data type." ));
+           }
+        }
+    }
 
-    const Vector3ui& overlap =
-                dataSourcePtr_->getVolumeInformation().overlap;
-    const size_t elemSize =
-                dataSourcePtr_->getVolumeInformation().getBytesPerVoxel();
-    const Vector3ui blockSize =
-                lodNodePtr_->getBlockSize() + overlap * 2;
-    return blockSize.product() * elemSize;
-}
+    bool load()
+    {
+        const DataType dataType = _dataSource.getVolumeInformation().dataType;
+        switch( _gpuDataType )
+        {
+            case GL_UNSIGNED_BYTE:
+                return setTextureData< uint8_t >( dataType != DT_UINT8 );
+            case GL_FLOAT:
+                return setTextureData< float >( dataType != DT_FLOAT32 );
+            case GL_UNSIGNED_SHORT:
+                return setTextureData< uint16_t >( dataType != DT_UINT16 );
+        }
+        return false;
+    }
 
-size_t TextureDataObject::getCacheSize() const
-{
-    if( !isValid() )
-        return 0;
+    void unload()
+    {
+        _data->release();
 
-    return data_->getAllocSize();
-}
+        const NodeId nodeId( _dataObject.getId( ));
+           LBVERB << "Texture Data released: " << nodeId
+               << std::endl;
+    }
 
-ConstVolumeDataSourcePtr TextureDataObject::getDataSource() const
-{
-    return dataSourcePtr_;
-}
+    AllocMemoryUnitPtr _data;
+    TextureDataObject& _dataObject;
+    TextureDataCache& _dataCache;
+    VolumeDataSource& _dataSource;
+    uint32_t _gpuDataType;
```

Comment:
The members are public (in a private class). When used from the facade, you already have ```_impl->member```, and we do actually have Impls with private members. I would prefer to have public members without underscore - after all, the methods don't have an underscore either.